src/resolvers/unixfs-v1/content/hamt-sharded-directory.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+const hamtShardedDirectoryContent = (cid, node, unixfs, path, resolve, ipld) => {
+  return async function * (options = {}) {
+    const links = node.links
+
+    for (const link of links) {
+      const name = link.name.substring(2)
+
+      if (name) {
+        const result = await resolve(link.cid, name, `${path}/${name}`, [], ipld)
+
+        yield result.entry
+      } else {
+        // descend into subshard
+        node = await ipld.get(link.cid)
+
+        for await (const file of hamtShardedDirectoryContent(link.cid, node, null, path, resolve, ipld)(options)) {
```

Comment:
Name the async generator function and call it here instead?

Reply:
Done.